qiskit_metal/toolbox_metal/layer_stack_handler.py

```diff
@@ -0,0 +1,347 @@
+# -*- coding: utf-8 -*-
+
+from textwrap import fill
+from xmlrpc.client import Boolean
+import pandas as pd
+from typing import List, Tuple
+from typing import Union
+from addict import Dict
+import os
+import logging
+
+from .parsing import parse_units, TRUE_STR, FALSE_STR
+
+
+class LayerStackHandler():
+    """Use DataFrame to hold information for multiple chips.
+
+    For all the chips, do NOT reuse the layer-number.  However, within the dataframe,
+    we can repeat layer numbers since a layer can have multiple datatypes.
+
+    """
+
+    Col_Names = [
+        'chip_name', 'layer', 'datatype', 'material', 'thickness', 'z_coord',
+        'fill'
+    ]
+
+    def __init__(self,
+                 multi_planar_design: 'MultiPlanar',
+                 fname: Union['str', None] = None) -> None:
+        """Use the information in MultiPlanar design to parse_value
+        and get the name of filename for layer stack.
+
+        Args:
+            multi_planar_design (MultiPlanar): Class that is child of QDesign.
+        """
+
+        self.multi_planar_design = multi_planar_design
+        self.logger = self.multi_planar_design.logger  # type: logging.Logger
+
+        self.filename_csv_df = None
+        if hasattr(self.multi_planar_design,
+                   'ls_filename') and self.multi_planar_design.ls_filename:
+            #populate pandas table from data in this file.
+            self.filename_csv_df = self.multi_planar_design.ls_filename
+        elif fname:
+            self.filename_csv_df = fname
+
+        self.ls_df = None
+        # self.column_names = [
+        #     'chip_name', 'layer', 'datatype', 'material', 'thickness',
+        #     'z_coord', 'fill'
+        # ]
+        self.layer_stack_default = Dict(chip_name=['main'],
+                                        layer=[1],
+                                        datatype=[0],
+                                        material=['pec'],
+                                        thickness=['2um'],
+                                        z_coord=['0um'],
+                                        fill=['true'])
+
+        self._init_dataframe()
+
+    def _init_dataframe(self) -> None:
+        """Must check if filename for layerstack is valid before trying to import to a pandas table.
+        """
+
+        if self.filename_csv_df:
+            #populate pandas table from data in this file.
+            abs_path = os.path.abspath(self.filename_csv_df)
+            if os.path.isfile(abs_path):
+                self._read_csv_df(abs_path)
+            else:
+                self.logger.error(
+                    f'Not able to read file.'
+                    f'File:{abs_path} not read. Check the name and path.')
+
+        else:
+            #enter very basic default data for pandas table.
+            self.ls_df = pd.DataFrame(data=self.layer_stack_default)
+
+    def get_properties_for_chip_layer_datatype(
```

Comment:
Given the restrictions of no repeated layers, is this method not the same really as 'get_properties_for_layer_datatype'?

Reply:
Yes, ASSUMING the chip will never be involved. What I can do is add the chip_name as a property to return.  I will delete this method and update the other method to include chip_name as output as part of properties.

Reply:
Merged, will resolve this comment.